### PR TITLE
Update validator for generic assay binary and categorical data

### LINKED ISSUE #7932: (Generic Assay) Backend support for Categorical data and Binary data
Backend Support:

- Update current [meta file format](https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#generic-assay-meta-file) to support two more type `CATEGORICAL` and `BINARY` as `DATATYPE`.
- Update data validator (maybe add controlled vocabulary, e.g. `yes/no` or `true/false` for `BINARY` type)

### PR DESCRIPTION
**Pull request by dippindots**
Fix #7932 
Only need to update validator for different data type, importer should use the same one.
- Rename previous validator to `GenericAssayContinuousValidator`
- Update validator for generic assay `binary` and `categorical` data
- Add unit tests for all three datatypes
- Update `File-Formats.md`

Ref: RFC can be found here: #7920

### CODE REVIEW

**Review by rmadupuri — APPROVED** [review]
The changes look good. Thank you @dippindots! We can merge the PR.

File: core/src/main/scripts/importer/validateData.py [hunk 1]
```diff
@@ -4361,56 +4358,113 @@ class GenericAssayValidator(GenericAssayWiseFileValidator):
...
+    """ Validator for files containing generic assay binary values.
+    """
+    def __init__(self, *args, **kwargs):
+        """Initialize the instance attributes of the data file validator."""
+        super(GenericAssayBinaryValidator, self).__init__(*args, **kwargs)
+
+    # (1) values defined in ALLOWED_VALUES
+    # (2) NA cell value is allowed; means value was not tested on a sample
+
+    ALLOWED_VALUES = ['yes', 'no', 'true', 'false'] + GenericAssayWiseFileValidator.NULL_VALUES
```
Comment (rmadupuri) [inline core/src/main/scripts/importer/validateData.py:4449]: Just a small question. Are the binary values fixed?

File: docs/File-Formats.md [hunk 2]
```diff
@@ -1298,15 +1298,15 @@ The cells contain the p-value for the GSVA score: A real number, between 0.0 and
...
+Generic Assay is a two dimensional matrix generalized to capture non-genetic measurements per sample. Instead of a gene per row and a sample per column, a Generic Assay file contains a generic entity per row and a sample per column. A generic entity is defined by the data curator and generally means something other than a gene. Some examples include, treatment response or mutational signatures. For each generic entity - sample pair, a (real number / text / binary value) represents a captured measurement.
 
 ### Generic Assay meta file
 The generic assay metadata file should contain the following fields:
 ```
 cancer_study_identifier: Same value as specified in meta file of the study
 genetic_alteration_type: GENERIC_ASSAY
 generic_assay_type: <GENERIC_ASSAY_TYPE>, e.g., "TREATMENT_RESPONSE" or "MUTATIONAL_SIGNATURE"
-datatype: LIMIT-VALUE
+datatype: value from LIMIT-VALUE / CATEGORICAL / BINARY
```
Comment (rmadupuri) [inline docs/File-Formats.md:1309]: Just to be consistent with how we define datatype in other profiles, can we update the line to `datatype: LIMIT-VALUE, CATEGORICAL or BINARY`?

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py:4449]: Yes, in the RFC, we planned to have a controlled vocabulary for binary type: https://docs.google.com/document/d/1-6O16_j5b5LeHA5SnChnlEKQTYhcwNh4AEwCxB8FwC8/edit?disco=AAAAHCuNjcs.

File: docs/File-Formats.md (hunk 2 above)
Comment (dippindots) [reply docs/File-Formats.md]: Sure, updated!

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (rmadupuri) [reply core/src/main/scripts/importer/validateData.py:4455]: Right. We can extend the list if we have to introduce other binary values.

File: docs/File-Formats.md (hunk 2 above)
Comment (rmadupuri) [reply docs/File-Formats.md]: Thanks!

Comment (rmadupuri) [reply docs/File-Formats.md]: The failed tests seems to be not related to the changes in this PR. Can we merge?

Comment (dippindots) [reply docs/File-Formats.md]: Yes, I think we can merge!

**Pull request merged**